### PR TITLE
Allow for listing view columns; use more specific hostname

### PR DESCRIPTION
**Pull request by jmcphers**
This change fixes two small issues with the viewer integration:

1) Previously, it was not possible to list the columns of a view object (`listColumns` only accepted a `table` argument). This is now possible; like `previewObject`, it now accepts either a `table` or a `view`.

2) Previously, the "host" for an ODBC connection was the database name. This has proven to be insufficiently specific; for instance, you might connect to the same database name on different servers, or with different usernames. The host key now includes both the username and server name (when known).

### CODE REVIEW

File: R/Viewer.R [hunk 1]
```diff
@@ -136,11 +136,54 @@ odbcListColumns <- function(connection, ...) {
   UseMethod("odbcListColumns")
 }
 
-odbcListColumns.OdbcConnection <- function(connection, table, catalog = NULL, schema = NULL, ...) {
+# given a connection, returns its "host name" (a unique string which identifies it)
+computeHostName <- function(connection) {
+
+  # start with the name of the database
+  host_name <- connection@info$dbname
```
Comment (jimhester) [inline R/Viewer.R]: I think this whole function can just be the following.

```r
paste(collapse = "_",
  string_values(
    connection@info$username, 
    connection@info$dbname,
    if (!identical(connection@info$servername, connection@info$dbname)) connection@info$servername
))
```

File: R/Viewer.R [hunk 2]
```diff
@@ -136,11 +136,54 @@ odbcListColumns <- function(connection, ...) {
...
+  if (!is.null(table) && !is.null(view)) {
+    stop("`table` and `view` can not both be used", call. = FALSE)
+  }
+
+  # Error if neither table and view are passed
+  if (is.null(table) && is.null(view)) {
+    stop("`table` and `view` can not both be `NULL`", call. = FALSE)
+  }
+
+  if (!is.null(table)) {
```
Comment (jimhester) [inline R/Viewer.R]: This conditional can be

`table %||% view`

File: R/Viewer.R [hunk 3]
```diff
@@ -286,7 +316,13 @@ on_connection_opened <- function(connection, code) {
 
   # use the database name as the display name
   display_name <- connection@info$dbname
```
Comment (jimhester) [inline R/Viewer.R]: I think all the following can be
```r
server_name <- paste(collapse = "@", string_values(user_name, server_name))

if (!identical(server_name, display_name)) {
  display_name <- paste(collapse = " - ", string_values(display_name, server_name)
}
```

And possibly just put in a `computeDisplayName()` function.

**Comment by jmcphers** [comment]
Rewritten to be more succinct (thanks!)

**Comment by jimhester** [comment]
Thanks! I merged this at https://github.com/rstats-db/odbc/commit/85a212a78e1114050bdff2275162979de2b09f06

**Pull request closed without merging**